lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,168 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
+        """
+        Draw the path with updated gc.
+        """
+        # Do not modify the input! Use copy instead.
+        gc0 = renderer.new_gc()
+        gc0.copy_properties(gc)
+
+        gc0 = self._update_gc(gc0, self._gc)
+        trans = affine + self._offset_transform(renderer)
+
+        thetar = -np.radians(self._angle)
+        trans_matrix = np.array([[np.cos(thetar), -np.sin(thetar)],
+                                 [np.sin(thetar), np.cos(thetar)]])
+
+        if self._flip:
+            iflip = -1
+        else:
+            iflip = 1
+
+        # Convert spacing parameter to pixels.
+        spcpx = renderer.points_to_pixels(self._spacing)
+
+        # Transform before evaluation because to_polygons works at resolution
+        # of one -- assuming it is working in pixel space.
+        transpath = affine.transform_path(tpath)
+
+        # Evaluate path to straight line segments that can be used to
+        # construct line ticks.
+        polys = transpath.to_polygons(closed_only=False)
+
+        for p in polys:
+            x = p[:, 0]
+            y = p[:, 1]
+
+            # Can not interpolate points or draw line if only one point in
+            # polyline.
+            if x.size < 2:
+                continue
+
+            # Find distance between points on the line
+            ds = np.hypot(x[1:] - x[:-1], + y[1:] - y[:-1])
+
+            # Build parametric coordinate along curve
+            s = np.concatenate(([0.0], np.cumsum(ds)))
+            stot = s[-1]
+
+            num = int(np.ceil(stot / spcpx))-1
+            # Pick parameter values for ticks.
+            s_tick = np.linspace(spcpx/2, stot-spcpx/2, num)
+
+            # Find points along the parameterized curve
+            x_tick = np.interp(s_tick, s, x)
+            y_tick = np.interp(s_tick, s, y)
+
+            # Find unit vectors in local direction of curve
+            delta_s = self._spacing * .001
+            u = (np.interp(s_tick + delta_s, s, x) - x_tick) / delta_s
+            v = (np.interp(s_tick + delta_s, s, y) - y_tick) / delta_s
+
+            # Normalize slope into unit slope vector.
+            n = np.hypot(u, v)
+            mask = n == 0
+            n[mask] = 1.0
+
+            uv = np.array([u / n, v / n]).T
+            uv[mask] = np.array([0, 0]).T
+
+            # Rotate and scale unit vector into tick vector
+            dxy = np.dot(uv, trans_matrix) * self._length * spcpx
+
+            # Build tick endpoints
+            x_end = x_tick + iflip * dxy[:, 0]
+            y_end = y_tick + iflip * dxy[:, 1]
+
+            # Interleave ticks to form Path vertices
+            xyt = np.empty((2 * num, 2), dtype=x_tick.dtype)
+            xyt[0::2, 0] = x_tick
+            xyt[1::2, 0] = x_end
+            xyt[0::2, 1] = y_tick
+            xyt[1::2, 1] = y_end
+
+            # Transform ticks back to data space before plotting
+            xy = affine.inverted().transform(xyt)
```

Comment:
Can you pass a lower transform and avoid transforming back?

Reply:
I don't have the slightest idea.  It took considerable effort to figure out how to make this work -- I am not particularly familiar with the internals of MatPlotLib.  I won't claim that there is not a better way to achieve this.

Reply:
The simplest change is that you could pass `affine.inverted() + trans` instead, which seems to work.

Reply:
What would I be passing that to?  I guess I don't know what 'pass a lower transform' means to this code.

Reply:
To `renderer.draw_path` below, instead of `trans`.

Reply:
Thanks for being explicit.

I've made the change below in 1b84421c3.

Personally, I don't think it is much of an improvement.  Of course, I am thinking about things in terms of the way I figured out how to make it work, so transforming into and out of pixel space was very explicit and deliberate to me.  If this is more 'the matplotlib way', I certainly won't argue.

Reply:
I'm assuming here that a transform calculation (i.e., a 3x3 matrix multiply) is much cheaper than transforming the path twice (something that depends on path length.)